src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -216,6 +216,10 @@ private function normalizeDefaults(array $defaults): array
             }
         }
 
+        if (true !== ($defaults['stateless'] ?? null)) {
+            @trigger_error('Not setting the "api_platform.defaults.stateless" configuration option to `true` is deprecated since API Platform 2.6 and it will default to `true` in 3.0. You can override this at the operation level if you have stateful operations (highly not recommended).', E_USER_DEPRECATED);
```

Comment:
It shouldn’t be forced to be true IMHO. Some (most ?) authorization mechanisms supported by Symfony require to use the session, and - even if using stateless mechanisms is a best practice - this must continue to be supported. It should be possible to disable the stateless mode (but it’s ok to have it enabled by default).

Reply:
I totally agree with you, I'll remove it then.

Reply:
@mtarld We must still trigger a deprecation error, however not forcing it to be set to `true`.

Reply:
Ok! Which message do you want to trigger then?

Reply:
Ok, then the new message will be:
`Not setting the "api_platform.defaults.stateless" configuration is deprecated since API Platform 2.6 and it will default to `true` in 3.0. You can override this at the operation level if you have stateful operations (highly not recommended).`